channeldb/graph.go

```diff
@@ -2107,6 +2112,45 @@ func (c *ChannelGraph) FilterKnownChanIDs(chansInfo []ChannelUpdateInfo,
 
 	var newChanIDs []uint64
 
+	// isStale returns whether the timestamp is too far into the past.
+	isStale := func(timestamp time.Time) bool {
+		return time.Since(timestamp) > ChannelUpdateExpiryPeriod
+	}
+
+	// isSkewed returns whether the timestamp is too far into the future.
+	isSkewed := func(timestamp time.Time) bool {
+		return time.Until(timestamp) > ChannelUpdateExpiryPeriod
+	}
+
+	// Sort out all channels with outdated or skewed timestamps. Both
+	// update msges need to be out of boundaries for us to skip it.
+	var cleanedChansInfo []ChannelUpdateInfo
+	for _, info := range chansInfo {
```

Comment:
think this needs to be conditional since the `ChannelUpdateInfo` won't have the timestamps set if the message didn't include them?

Reply:
good catch, yes will move this check into the gossiper.